executor/builder.go

```diff
@@ -856,29 +862,16 @@ func (b *executorBuilder) buildTableReader(v *plan.PhysicalTableReader) Executor
 		ranges:    ts.Ranges,
 	}
 
-	for _, col := range v.Schema().Columns {
-		dagReq.OutputOffsets = append(dagReq.OutputOffsets, uint32(col.Index))
+	for i := range v.Schema().Columns {
+		dagReq.OutputOffsets = append(dagReq.OutputOffsets, uint32(i))
```

Comment:
For example, if index is a_b_c and the SQL is `select c from t where a = 1`, we only need a, c but the Index Scan in KV will decode all of them. So when KV are returning, it will pick a and c from a,b,c instead of picking all of them. So we should resolve indices of a, c as 1, 3 but 1, 2